src/lockscreen.js

```diff
@@ -0,0 +1,81 @@
+/* exported Lockscreen */
+
+const {GObject, Gtk} = imports.gi;
+
+const _css = `
+.lockscreen {
+    background: url('/usr/share/icons/Adwaita/256x256/status/dialog-password.png'),
+                linear-gradient(to right, rgba(61, 70, 81, 1) 0%,
+                                          rgba(61, 70, 81, 1) 50%,
+                                          rgba(61, 70, 81, 0) 51%,
+                                          rgba(61, 70, 81, 0) 100%),
+                linear-gradient(to right, rgba(61, 70, 81, 0) 0%,
+                                          rgba(61, 70, 81, 0) 50%,
+                                          rgba(61, 70, 81, 1) 51%,
+                                          rgba(61, 70, 81, 1) 100%);
+    background-repeat: no-repeat;
+    background-position: center, -600px, 600px;
+    opacity: 0;
+    transition-property: background-position, opacity;
+    transition-duration: 0.5s, 0.75s;
+    transition-timing-function: cubic-bezier(.5, 0, 1, .5), linear;
+}
+
+.lockscreen.locked {
+    background-position: center, 0px, 0px;
+    opacity: 1;
+}
+`;
+
+var Lockscreen = GObject.registerClass({
+    Properties: {
+        locked: GObject.ParamSpec.boolean('locked', 'Locked', '',
+            GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+            true),
+    },
+}, class Lockscreen extends Gtk.Overlay {
+    _init(props = {}) {
+        super._init(props);
+
+        this._overlay = new Gtk.Frame({
+            expand: true,
+            visible: true,
+        });
+        this.add_overlay(this._overlay);
+
+        const style = this._overlay.get_style_context();
+        style.add_class('lockscreen');
```

Comment:
Instead of a style class, you should be able to set a CSS name for this class; gjs has a way to do that directly as part of the class definition.

Reply:
I want the style class on the overlay widget, instead of the class itself. However, since the style provider only applies to the one widget anyway, I'll just use its existing CSS name (`frame`).